biblib/views/operations_view.py

```diff
@@ -203,33 +197,60 @@ def post(self, library):
             current_app.logger.error('Wrong type passed for POST: {0} [{1}]'
                                      .format(request.data, error))
             return err(WRONG_TYPE_ERROR)
-
-        if data['action'] in ['union', 'intersection', 'difference']:
-            if 'libraries' not in data:
-                return err(NO_LIBRARY_SPECIFIED_ERROR)
-            if 'name' not in data:
-                data['name'] = 'Untitled {0}.'.format(get_date().isoformat())
-            if 'public' not in data:
-                data['public'] = False
-
-        if data['action'] == 'copy':
-            if 'libraries' not in data:
-                return err(NO_LIBRARY_SPECIFIED_ERROR)
-            if len(data['libraries']) > 1:
-                return err(TOO_MANY_LIBRARIES_SPECIFIED_ERROR)
-
+       
+        has_read_access_primary = False
+        has_read_access_secondary = False 
+        has_write_access_secondary = False
+        primary_is_public = False 
+        secondary_is_public = False 
+        
         lib_names = []
+
         with current_app.session_scope() as session:
             primary = session.query(Library).filter_by(id=library_uuid).one()
             lib_names.append(primary.name)
-            if 'libraries' in data:
-                for lib in data['libraries']:
-                    try:
-                        secondary_uuid = self.helper_slug_to_uuid(lib)
-                    except TypeError:
-                        return err(BAD_LIBRARY_ID_ERROR)
-                    secondary = session.query(Library).filter_by(id=secondary_uuid).one()
-                    lib_names.append(secondary.name)
+            if primary.public: primary_is_public = True
+            if self.read_access(service_uid=user_editing_uid,
+                                        library_id=library_uuid):
+                has_read_access_primary = True 
+        
+            for lib in data.get('libraries', []):
```

Comment:
I think this might miss a corner case where if someone specifies a mix of libraries with various access types. This isn't an issue for the `copy` method because we limit the number of secondary libraries to 1, but for other methods, they could potentially acquire access to a secondary library they are not supposed to have (ie. a user could compute the union with three libraries where they only have access to two of them and gain access to the documents stored in the third library.) I have to double-check all the methods but a simple first past fix might be to raise an exception immediately if a user has no permissions on a private secondary library.

Reply:
Thank you for catching that. I had changed the logic a little bit and forgot to account for the implications here. Good one. I'll fix it right away.